src/control/projects/mod.rs

```diff
@@ -1,76 +1,54 @@
-use std::fmt::Debug;
-
-use axum::{
-    extract::Path,
-    headers::{authorization::Bearer, Authorization},
-    Extension, Json, TypedHeader,
-};
+use axum::{extract::Path, Extension, Json, TypedHeader};
 use http::StatusCode;
 use tracing::info;
 
 use crate::{
     domain::projects::{Project, ProjectData, ProjectMetaData},
-    repository::projects::{ProjectGetError, ProjectPostError, ProjectRepository},
-    repository::{
-        crud::{CrudCheckError, CrudRepository},
-        projects::ProjectPutError,
-    },
+    domain::headers::XUserId,
+    repository::projects::ProjectUpdateError,
+    repository::projects::{ProjectGetError, ProjectInsertError, ProjectRepository},
 };
 
 #[tracing::instrument(skip_all)]
-pub async fn get_projects<T: ProjectRepository + CrudRepository + Clone + Send + Sync>(
+pub async fn get_projects<T: ProjectRepository + Clone + Send + Sync>(
     Extension(repository): Extension<T>,
-    TypedHeader(user_id): TypedHeader<Authorization<Bearer>>,
+    TypedHeader(user_id): TypedHeader<XUserId>
```

Comment:
Can't we use more even more pattern matching (`TypedHeader(XUserId(user_id))`) to immediately get the value instead?

Reply:
Sure thing.